core/includes/bootstrap.inc

```diff
@@ -1995,6 +2008,50 @@ function request_uri() {
 }
 
 /**
+ * Logs a deprecation warning.
+ *
+ * This is a wrapper function for watchdog() to record uses of deprecated
+ * functions in Backdrop modules. To prevent an explosion of WATCHDOG_DEPRECATED
+ * messages, only the first use of a function in a particular file:line will get
+ * a watchdog message.
+ *
+ * @param $type
+ *   The category to which this message belongs.
+ * @param $func
+ *   The function or method deprecated.  Usually, __FUNCTION__ or __METHOD__.
+ * @param $link
+ *   A link to associate with the message.
+ *
+ * @see watchdog()
+ */
+function watchdog_deprecated_function($type, $func, $link=NULL) {
+  static $deprecated_calls = array();
+
```

Comment:
Could we duplicate some of the ` config('system.core')->get('watchdog_enabled_severity_levels');` at this point? `debug_backtrace()` pulls a huge amount of data and looping through it could have a performance impact. If logging is disabled in the first place, it would be preferable to avoid the processing if we're just going to throw the result away.